json_schema_fuzz/__init__.py

```diff
@@ -1,15 +1,108 @@
 """JSON schema fuzzer."""
+import json
 import random
 import string
+from decimal import Decimal
 
 import exrex
 
 from .schema_operations import merge
+from .utils import ALL_TYPES, listify, random_multiple_in_range
+
+MAX_REJECTED_SAMPLES = 1000
+
+
+class RejectionSamplingFailed(Exception):
+    """
+    Failed to generate sample that satisfies all criteria
+    """
 
 
 def random_integer(schema):
     """Generate random integer."""
-    return random.randint(0, 10)
+
+    multiple_of = schema.get("multipleOf", Decimal(1))
+
+    default_min = Decimal(1 * multiple_of)
+    default_max = Decimal(100 * multiple_of)
+
+    minimum = max(
+        schema.get("minimum", default_min),
+        schema.get("exclusiveMinimum", default_min) + 1
+    )
+    maximum = min(
+        schema.get("maximum", default_max),
+        schema.get("exclusiveMaximum", default_max) - 1
+    )
+
+    not_multiple_of = schema.get("notMultipleOf", [])
+    if not isinstance(not_multiple_of, list):
+        not_multiple_of = [not_multiple_of]
+
+    for _ in range(MAX_REJECTED_SAMPLES):
+        # Generate new value
+        value = random_multiple_in_range(
+            minimum,
+            maximum,
+            multiple_of
+        )
+
+        # Verify
+        is_multiple_of = [value % num == 0 for num in not_multiple_of]
+        if any(is_multiple_of):
+            continue
+        return int(value)
+    raise RejectionSamplingFailed()
+
+
+def random_number(schema):
+    """Generate random number."""
+
+    decimal_digits = 4
+    smallest_interval = 10 ** Decimal(-decimal_digits)
+
+    multiple_of = schema.get("multipleOf", smallest_interval)
+
+    if "minimum" in schema or "exclusiveMinimum" in schema:
+        minimum = max(
+            schema.get("minimum", Decimal("-Infinity")),
+            schema.get("exclusiveMinimum", Decimal("-Infinity"))
+            + smallest_interval
+        )
+    else:
+        minimum = Decimal(1) * multiple_of
+
+    if "maximum" in schema or "exclusiveMaximum" in schema:
+        maximum = min(
+            schema.get("maximum", Decimal("Infinity")),
+            schema.get("exclusiveMaximum", Decimal("Infinity"))
+            - smallest_interval
+        )
+    else:
+        maximum = Decimal(100) * multiple_of
+
+    # Range can flip if multiple_of is <1
+    # so we sort it
+    minimum, maximum = sorted((minimum, maximum))
```

Comment:
Will this cause some schemas to be (wrongly) satisfiable if they start out with minimum > maximum?